pkg/metrics/metrics.go

```diff
@@ -127,3 +149,13 @@ func registerMetrics() error {
 func GetRoundTripper() http.RoundTripper {
 	return roundTripper
 }
+
+// RegisterCustomResource increases the count of the custom_resource_total metric
+func RegisterCustomResource(resourceType string) {
+	customResourceGaugeVec.With(prometheus.Labels{ResourceTypeLabel: resourceType}).Inc()
+}
+
+// DeregisterCustomResource decreases the count of the custom_resource_total metric
+func DeregisterCustomResource(resourceType string) {
```

Comment:
Can this be named DecCustomResource since "Register" has it's own meaning in the prometheus client?